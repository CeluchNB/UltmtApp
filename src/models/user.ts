import { Realm } from '@realm/react'

export const DisplayUserSchema: Realm.ObjectSchema = {
    name: 'DisplayUser',
    embedded: true,
    properties: {
        _id: 'string',
        firstName: 'string',
        lastName: 'string',
        username: 'string',
        localGuest: 'bool?',
    },
}
