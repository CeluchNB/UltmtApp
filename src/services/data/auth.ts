import * as Constants from '../../utils/constants'
import { ApiError } from '../../types/services'
import { AxiosResponse } from 'axios'
import EncryptedStorage from 'react-native-encrypted-storage'
import jwt_decode from 'jwt-decode'
import { isTokenExpired, throwApiError } from '../../utils/service-utils'
import {
    login as networkLogin,
    logout as networkLogout,
    refreshToken as networkRefreshToken,
} from '../network/auth'

/**
 * Method to login
 * @param username username or email
 * @param password password
 * @returns user token
 * @throws error if backend returns an error
 */
export const login = async (
    username: string,
    password: string,
): Promise<void> => {
    try {
        const response = await networkLogin(username, password)

        const tokens = response.data.tokens
        const { access, refresh } = tokens
        await EncryptedStorage.setItem('access_token', access)
        await EncryptedStorage.setItem('refresh_token', refresh)
    } catch (error: any) {
        return throwApiError(error, Constants.GENERIC_LOGIN_ERROR)
    }
}

/**
 * Logout current user
 * @param token token of current user
 * @throws error if backend returns an error
 */
export const logout = async () => {
    try {
        await withToken(networkLogout)
        await EncryptedStorage.removeItem('access_token')
        await EncryptedStorage.removeItem('refresh_token')
    } catch (error: any) {
        await EncryptedStorage.removeItem('access_token')
        await EncryptedStorage.removeItem('refresh_token')
        throw throwApiError(error, Constants.GENERIC_LOGOUT_ERROR)
    }
}

/**
 * Method to refresh the user's authentication token
 * @returns the new access token
 */
export const refreshToken = async (): Promise<string> => {
    try {
        const token = await EncryptedStorage.getItem('refresh_token')
        if (!token) {
            throw new ApiError(Constants.GENERIC_GET_TOKEN_ERROR)
        }
        const response = await networkRefreshToken(token)
        const { tokens } = response.data
        const { access, refresh } = tokens
        await EncryptedStorage.setItem('access_token', access)
        await EncryptedStorage.setItem('refresh_token', refresh)
        return access
    } catch (error: any) {
        await EncryptedStorage.removeItem('access_token')
        await EncryptedStorage.removeItem('refresh_token')
        throw throwApiError(error, Constants.GENERIC_GET_TOKEN_ERROR)
    }
}

/**
 * Method to make an API call with an authentication token that could be expired.
 * Will make the refresh call and try again if the token is expired.
 * @param networkCall method to call
 * @param args arguments of method
 * @returns expected response of network call
 */
export const withToken = async (
    networkCall: (token: string, ...args: any) => Promise<AxiosResponse>,
    ...args: any
): Promise<AxiosResponse> => {
    try {
        const currentToken =
            (await EncryptedStorage.getItem('access_token')) || ''
        const response = await networkCall(currentToken, ...args)
        return response
    } catch (error: any) {
        const errorJson = error.toJSON()
        if (errorJson.status !== 401) {
            throw error
        }
        try {
            const newToken = await refreshToken()
            return await networkCall(newToken, ...args)
        } catch (error2) {
            throw throwApiError(error2, Constants.GENERIC_GET_TOKEN_ERROR)
        }
    }
}

/**
 * Determine if a user has a valid refresh token
 * @returns true if user logged in
 * @throws error if user is not currently logged in
 */
export const isLoggedIn = async (): Promise<boolean> => {
    try {
        const rToken = await EncryptedStorage.getItem('refresh_token')
        if (!rToken) {
            throw new ApiError(Constants.GENERIC_GET_TOKEN_ERROR)
        }

        const { exp: rExp } = jwt_decode(rToken) as any
        if (isTokenExpired(rExp)) {
            throw new ApiError(Constants.GENERIC_GET_TOKEN_ERROR)
        }

        // TODO: can probably delete this
        const accessToken = await EncryptedStorage.getItem('access_token')
        if (!accessToken) {
            throw new ApiError(Constants.GENERIC_GET_TOKEN_ERROR)
        }
        return true
    } catch (error) {
        throw throwApiError(error, Constants.GENERIC_GET_TOKEN_ERROR)
    }
}

/**
 * Method to determine if a user has a valid access token
 * @returns boolean if user has valid access token
 */
export const hasValidAccessToken = async (): Promise<boolean> => {
    const token = await EncryptedStorage.getItem('access_token')
    if (!token) {
        return false
    }

    const { exp } = jwt_decode(token) as any
    return !isTokenExpired(exp)
}

/**
 * Method to refresh the user's token if necessary. Used for live actions
 * since sockets give no response to determine if a reauth is necessary.
 */
export const refreshTokenIfNecessary = async () => {
    const hasToken = await hasValidAccessToken()
    if (!hasToken) {
        await refreshToken()
    }
}
