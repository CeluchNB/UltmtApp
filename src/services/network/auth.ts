import { ApiResponse } from '../../types/services'
import EncryptedStorage from 'react-native-encrypted-storage'
import { API_KEY, API_URL_V1 } from 'react-native-dotenv'
import axios, { AxiosResponse } from 'axios'

export const login = async (
    username: string,
    password: string,
): Promise<AxiosResponse> => {
    return await axios.post(
        `${API_URL_V1}/auth/login`,
        {
            password,
            email: username,
        },
        {
            headers: {
                'X-API-Key': API_KEY,
            },
        },
    )
}

export const logout = async (token: string): Promise<AxiosResponse> => {
    return await axios.post(
        `${API_URL_V1}/auth/logout`,
        {},
        {
            headers: {
                Authorization: `Bearer ${token}`,
                'X-API-Key': API_KEY,
            },
        },
    )
}

export const refreshToken = async (token: string): Promise<AxiosResponse> => {
    return await axios.post(
        `${API_URL_V1}/auth/refresh`,
        {},
        {
            headers: {
                Authorization: `Bearer ${token}`,
                'X-API-Key': API_KEY,
            },
        },
    )
}

export const getAccessToken = async (): Promise<ApiResponse> => {
    try {
        const token = await EncryptedStorage.getItem('access_token')
        if (token) {
            return { data: token }
        }
    } catch (error) {
        throw error
    }
    throw new Error('No token available')
}
