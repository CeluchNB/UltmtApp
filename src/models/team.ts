import { DisplayUser } from '../types/user'
import { Realm } from '@realm/react'
import { LocalUser, Team } from '../types/team'

export const DisplayTeamSchema: Realm.ObjectSchema = {
    name: 'DisplayTeam',
    embedded: true,
    properties: {
        _id: 'string',
        place: 'string',
        name: 'string',
        teamname: 'string',
        seasonStart: 'string',
        seasonEnd: 'string',
    },
}

export const GuestTeamSchema: Realm.ObjectSchema = {
    name: 'GuestTeam',
    embedded: true,
    properties: {
        _id: 'string?',
        place: 'string?',
        name: 'string',
        teamname: 'string?',
        seasonStart: 'string?',
        seasonEnd: 'string?',
    },
}

export class TeamSchema {
    static schema: Realm.ObjectSchema = {
        name: 'Team',
        primaryKey: '_id',
        properties: {
            _id: 'string',
            place: 'string',
            name: 'string',
            teamname: 'string',
            managers: { type: 'list', objectType: 'DisplayUser' },
            players: { type: 'list', objectType: 'DisplayUser' },
            seasonStart: 'string',
            seasonEnd: 'string',
            seasonNumber: 'int',
            continuationId: 'string',
            rosterOpen: 'bool',
        },
    }

    _id: string
    place: string
    name: string
    teamname: string
    managers: DisplayUser[]
    players: LocalUser[]
    seasonStart: string
    seasonEnd: string
    seasonNumber: number
    continuationId: string
    rosterOpen: boolean

    constructor(team: Team) {
        this._id = team._id
        this.place = team.place
        this.name = team.name
        this.teamname = team.teamname
        this.managers = team.managers
        this.players = team.players
        this.seasonStart = team.seasonStart
        this.seasonEnd = team.seasonEnd
        this.seasonNumber = team.seasonNumber
        this.continuationId = team.continuationId
        this.rosterOpen = team.rosterOpen
    }
}
