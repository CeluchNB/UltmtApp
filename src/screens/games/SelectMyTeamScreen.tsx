import * as Constants from '../../utils/constants'
import { AllScreenProps } from '../../types/navigation'
import { AppDispatch } from '../../store/store'
import BaseScreen from '../../components/atoms/BaseScreen'
import { DisplayTeam } from '../../types/team'
import PrimaryButton from '../../components/atoms/PrimaryButton'
import React from 'react'
import ScreenTitle from '../../components/atoms/ScreenTitle'
import TeamListItem from '../../components/atoms/TeamListItem'
import { fetchProfile } from '../../store/reducers/features/account/accountReducer'
import { isLoggedIn } from '../../services/data/auth'
import { size } from '../../theme/fonts'
import {
    ActivityIndicator,
    FlatList,
    StyleSheet,
    Text,
    View,
} from 'react-native'
import {
    selectFetchProfileLoading,
    selectManagerTeams,
} from '../../store/reducers/features/account/accountReducer'
import { useColors, useData } from '../../hooks'
import { useDispatch, useSelector } from 'react-redux'

const SelectMyTeamScreen: React.FC<AllScreenProps> = ({ navigation }) => {
    const { colors } = useColors()
    const dispatch = useDispatch<AppDispatch>()
    const managerTeams = useSelector(selectManagerTeams)
    const fetchProfileLoading = useSelector(selectFetchProfileLoading)

    const { data: isAuth, loading, refetch } = useData(isLoggedIn)
    React.useEffect(() => {
        const unsubscribe = navigation.addListener('focus', () => {
            refetch()
        })

        return unsubscribe
    })

    React.useEffect(() => {
        if (isAuth && managerTeams.length < 1) {
            dispatch(fetchProfile())
        }
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [isAuth])

    const onSelect = (teamOne: DisplayTeam) => {
        navigation.navigate('SelectOpponent', { teamOne })
    }

    const onCreateTeam = async () => {
        navigation.push('Tabs', {
            screen: 'Account',
            params: { screen: 'CreateTeam' },
        })
    }

    const onLogin = async () => {
        navigation.navigate('Tabs', {
            screen: 'Account',
            params: { screen: 'Login' },
        })
    }

    const styles = StyleSheet.create({
        title: {
            marginTop: 10,
        },
        list: {
            marginTop: 20,
        },
        error: {
            color: colors.gray,
            fontSize: size.fontLarge,
        },
    })

    return (
        <BaseScreen containerWidth="80%">
            <ScreenTitle style={styles.title} title="Select My Team" />
            {(loading || fetchProfileLoading) && (
                <ActivityIndicator
                    color={colors.textPrimary}
                    testID="select-team-loading"
                />
            )}
            {!loading && !isAuth && (
                <View>
                    <Text style={styles.error}>{Constants.AUTH_TO_CREATE}</Text>
                    <PrimaryButton
                        text="log in"
                        onPress={onLogin}
                        loading={false}
                    />
                </View>
            )}
            {managerTeams.length < 1 && isAuth && !fetchProfileLoading && (
                <View>
                    <Text style={styles.error}>
                        {Constants.MANAGE_TO_CREATE}
                    </Text>
                    <PrimaryButton
                        text="create team"
                        loading={false}
                        onPress={onCreateTeam}
                    />
                </View>
            )}
            {managerTeams.length > 0 && (
                <FlatList
                    style={styles.list}
                    data={managerTeams}
                    renderItem={({ item }) => {
                        return (
                            <TeamListItem
                                team={item}
                                onPress={async () => {
                                    onSelect(item)
                                }}
                            />
                        )
                    }}
                />
            )}
        </BaseScreen>
    )
}

export default SelectMyTeamScreen
