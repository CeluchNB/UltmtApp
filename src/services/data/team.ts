import * as Constants from '../../utils/constants'
import { ApiError } from '../../types/services'
import { throwApiError } from '../../utils/service-utils'
import { CreateTeam, Team } from '../../types/team'
import {
    createTeam as networkCreateTeam,
    getManagedTeam as networkGetManagedTeam,
    getTeam as networkGetTeam,
    removePlayer as networkRemovePlayer,
    rollover as networkRollover,
    searchTeam as networkSearchTeam,
    toggleRosterStatus as networkToggleRosterStatus,
} from '../network/team'

/**
 * Method to create team
 * @param token token of user creating team
 * @param data create team data
 * @returns created team
 * @throws error if team is not created successfully
 */
export const createTeam = async (
    token: string,
    data: CreateTeam,
): Promise<Team> => {
    try {
        const response = await networkCreateTeam(token, data)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.CREATE_TEAM_ERROR)
    }
}

/**
 * Method to search for teams by place, name, or username
 * @param term search term
 * @returns list of teams matching term
 * @throws error if backend returns an error
 */
export const searchTeam = async (term: string): Promise<Team[]> => {
    try {
        if (term.length < 3) {
            throw new ApiError('Not enough characters to search.')
        }
        const response = await networkSearchTeam(term)
        return response.data
    } catch (error) {
        return throwApiError(error, Constants.SEARCH_ERROR)
    }
}

/**
 * Method to get team that the current user manages
 * @param token jwt of current user
 * @param id team id
 * @returns team object
 * @throws error if backend returns error
 */
export const getManagedTeam = async (
    token: string,
    id: string,
): Promise<Team> => {
    try {
        const response = await networkGetManagedTeam(token, id)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.GET_TEAM_ERROR)
    }
}

/**
 * Method to get a public team
 * @param id id of team to get
 * @returns team object
 * @throws error if backend returns an error
 */
export const getTeam = async (id: string): Promise<Team> => {
    try {
        const response = await networkGetTeam(id)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.GET_TEAM_ERROR)
    }
}

/**
 * Method to change a team's roster status
 * @param token jwt of user changing team
 * @param id id of team to update
 * @param open boolean for team's roster status
 * @returns updated team object
 * @throws error if backend returns error
 */
export const toggleRosterStatus = async (
    token: string,
    id: string,
    open: boolean,
): Promise<Team> => {
    try {
        const response = await networkToggleRosterStatus(token, id, open)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.EDIT_TEAM_ERROR)
    }
}

/**
 * Method to remove a player from a team
 * @param token token of current user
 * @param teamId team player is on
 * @param userId id of player to remove
 * @returns updated team object
 * @throws error if backend returns an error
 */
export const removePlayer = async (
    token: string,
    teamId: string,
    userId: string,
): Promise<Team> => {
    try {
        const response = await networkRemovePlayer(token, teamId, userId)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.EDIT_TEAM_ERROR)
    }
}

/**
 * Method to start a new season for a team. After this method is called,
 * the old team is no longer editable.
 * @param token jwt of manager of team
 * @param teamId id of team to move over
 * @param copyPlayers boolean for deciding if current roster should be kept
 * @param seasonStart start year for season
 * @param seasonEnd start end for season
 * @returns newly created team object
 * @throws error if backend returns an error
 */
export const rollover = async (
    token: string,
    teamId: string,
    copyPlayers: boolean,
    seasonStart: string,
    seasonEnd: string,
): Promise<Team> => {
    try {
        const response = await networkRollover(
            token,
            teamId,
            copyPlayers,
            seasonStart,
            seasonEnd,
        )
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.EDIT_TEAM_ERROR)
    }
}
