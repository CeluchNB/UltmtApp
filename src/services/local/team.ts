import { Realm } from '@realm/react'
import { Team } from '../../types/team'
import { TeamSchema } from '../../models'
import { getRealm } from '../../models/realm'

export const saveTeams = async (teams: Team[]) => {
    const realm = await getRealm()
    realm.write(() => {
        for (const team of teams) {
            realm.create('Team', team, Realm.UpdateMode.Modified)
        }
    })
}

export const getManagingTeams = async (id: string): Promise<Team[]> => {
    const realm = await getRealm()
    const teams = await realm.objects<TeamSchema>('Team')
    return teams
        .filter(t => t.managers.map(m => m._id).includes(id))
        .map(t => parseTeam(t))
}

const parseTeam = (schema: TeamSchema): Team => {
    return JSON.parse(
        JSON.stringify({
            _id: schema._id,
            place: schema.place,
            name: schema.name,
            teamname: schema.teamname,
            managers: schema.managers,
            players: schema.players,
            seasonStart: schema.seasonStart,
            seasonEnd: schema.seasonEnd,
            seasonNumber: schema.seasonNumber,
            continuationId: schema.continuationId,
            rosterOpen: schema.rosterOpen,
            requests: [],
            games: [],
        }),
    )
}
