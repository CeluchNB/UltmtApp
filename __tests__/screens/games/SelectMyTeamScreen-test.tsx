import * as AuthData from '../../../src/services/data/auth'
import * as Constants from '../../../src/utils/constants'
import * as TeamData from '../../../src/services/data/team'
import * as UserData from '../../../src/services/data/user'
import { NavigationContainer } from '@react-navigation/native'
import { Provider } from 'react-redux'
import React from 'react'
import { SelectMyTeamProps } from '../../../src/types/navigation'
import SelectMyTeamScreen from '../../../src/screens/games/SelectMyTeamScreen'
import { Team } from '../../../src/types/team'
import { fetchProfileData } from '../../../fixtures/data'
import { setProfile } from '../../../src/store/reducers/features/account/accountReducer'
import store from '../../../src/store/store'
import {
    fireEvent,
    render,
    waitFor,
    waitForElementToBeRemoved,
} from '@testing-library/react-native'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

const navigate = jest.fn()
const addListener = jest.fn()
const push = jest.fn()

const props: SelectMyTeamProps = {
    navigation: {
        navigate,
        addListener,
        push,
    } as any,
    route: {} as any,
}

describe('SelectMyTeamScreen', () => {
    afterEach(() => {
        jest.resetAllMocks()
    })

    it('should match snapshot with unauthenticated user', async () => {
        jest.spyOn(AuthData, 'isLoggedIn').mockReturnValue(
            Promise.resolve(false),
        )
        const snapshot = render(
            <Provider store={store}>
                <NavigationContainer>
                    <SelectMyTeamScreen {...props} />
                </NavigationContainer>
            </Provider>,
        )

        await waitForElementToBeRemoved(() =>
            snapshot.getByTestId('select-team-loading'),
        )

        expect(snapshot.toJSON()).toMatchSnapshot()
        expect(snapshot.getByText(Constants.AUTH_TO_CREATE)).not.toBeNull()
        fireEvent.press(snapshot.getByText('log in'))
        expect(navigate).toHaveBeenCalledWith('Tabs', {
            screen: 'Account',
            params: { screen: 'Login' },
        })
    })

    it('should match snapshot without manager teams', async () => {
        jest.spyOn(AuthData, 'isLoggedIn').mockReturnValue(
            Promise.resolve(true),
        )
        jest.spyOn(UserData, 'fetchProfile').mockReturnValueOnce(
            Promise.resolve({ ...fetchProfileData, managerTeams: [] }),
        )
        const snapshot = render(
            <Provider store={store}>
                <NavigationContainer>
                    <SelectMyTeamScreen {...props} />
                </NavigationContainer>
            </Provider>,
        )

        await waitForElementToBeRemoved(() =>
            snapshot.getByTestId('select-team-loading'),
        )

        expect(snapshot.toJSON()).toMatchSnapshot()
        expect(snapshot.getByText(Constants.MANAGE_TO_CREATE)).not.toBeNull()
        fireEvent.press(snapshot.getByText('create team'))
        expect(push).toHaveBeenCalledWith('Tabs', {
            screen: 'Account',
            params: { screen: 'CreateTeam' },
        })
    })

    it('should match snapshot with manager teams', async () => {
        jest.spyOn(AuthData, 'isLoggedIn').mockReturnValue(
            Promise.resolve(true),
        )
        jest.spyOn(UserData, 'fetchProfile').mockReturnValueOnce(
            Promise.resolve({ ...fetchProfileData, managerTeams: [] }),
        )
        jest.spyOn(TeamData, 'getTeam').mockReturnValueOnce(
            Promise.resolve({} as Team),
        )
        store.dispatch(
            setProfile({
                managerTeams: [
                    {
                        _id: 'team1',
                        place: 'Place',
                        name: 'Name',
                        teamname: 'place1name1',
                        seasonStart: '2022',
                        seasonEnd: '2022',
                    },
                ],
            }),
        )
        const snapshot = render(
            <Provider store={store}>
                <NavigationContainer>
                    <SelectMyTeamScreen {...props} />
                </NavigationContainer>
            </Provider>,
        )

        await waitForElementToBeRemoved(() =>
            snapshot.getByTestId('select-team-loading'),
        )

        expect(snapshot.toJSON()).toMatchSnapshot()
        expect(snapshot.getByText('@place1name1')).not.toBeNull()
        fireEvent.press(snapshot.getByText('@place1name1'))
        await waitFor(async () => {
            expect(navigate).toHaveBeenCalledWith('SelectOpponent', {})
        })
    })
})
