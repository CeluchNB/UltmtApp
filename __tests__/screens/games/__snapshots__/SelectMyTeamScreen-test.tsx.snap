// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with manager teams 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "width": "80%",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#ffd600",
            "fontSize": 61.64383561643835,
            "fontWeight": "bold",
          },
          Object {
            "marginTop": 10,
          },
        ]
      }
    >
      Select My Team
    </Text>
    <RCTScrollView
      data={
        Array [
          Object {
            "_id": "team1",
            "name": "Name",
            "place": "Place",
            "seasonEnd": "2022",
            "seasonStart": "2022",
            "teamname": "place1name1",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "marginTop": 20,
        }
      }
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#ababab",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Place Name
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#ffd600",
                      "fontSize": 30.821917808219176,
                      "fontWeight": "bold",
                    }
                  }
                >
                  @
                  place1name1
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#ffd600",
                      "fontSize": 20.54794520547945,
                      "fontWeight": "bold",
                    }
                  }
                >
                  2022
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ],
                ]
              }
              testID="go-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#ffd600",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ó°…‚
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`should match snapshot with unauthenticated user 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "width": "80%",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#ffd600",
            "fontSize": 61.64383561643835,
            "fontWeight": "bold",
          },
          Object {
            "marginTop": 10,
          },
        ]
      }
    >
      Select My Team
    </Text>
    <View>
      <Text
        style={
          Object {
            "color": "#ababab",
            "fontSize": 61.64383561643835,
          }
        }
      >
        Log in to create a game.
      </Text>
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#ffd600",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "minWidth": "auto",
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              log in
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`should match snapshot without manager teams 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "width": "80%",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#ffd600",
            "fontSize": 61.64383561643835,
            "fontWeight": "bold",
          },
          Object {
            "marginTop": 10,
          },
        ]
      }
    >
      Select My Team
    </Text>
    <View>
      <Text
        style={
          Object {
            "color": "#ababab",
            "fontSize": 61.64383561643835,
          }
        }
      >
        Manage a team to create a game.
      </Text>
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#ffd600",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "minWidth": "auto",
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              create team
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
